Common/Data/UniverseSelection/FuturesChainUniverse.cs

```diff
@@ -0,0 +1,196 @@
+                    ))
+                .Select(config => new SubscriptionRequest(
+                    isUniverseSubscription: false,
+                    universe: this,
+                    security: security,
+                    configuration: config,
+                    startTimeUtc: currentTimeUtc,
+                    endTimeUtc: maximumEndTimeUtc
+                    ));
+        }
+
+        /// <summary>
+        /// Creates and configures a security for the specified symbol
+        /// </summary>
+        /// <param name="symbol">The symbol of the security to be created</param>
+        /// <param name="algorithm">The algorithm instance</param>
+        /// <param name="marketHoursDatabase">The market hours database</param>
+        /// <param name="symbolPropertiesDatabase">The symbol properties database</param>
+        /// <returns>The newly initialized security object</returns>
+        public override Security CreateSecurity(Symbol symbol, IAlgorithm algorithm, MarketHoursDatabase marketHoursDatabase, SymbolPropertiesDatabase symbolPropertiesDatabase)
+        {
+            // set the underlying security and pricing model from the canonical security
+            var future = (Future)base.CreateSecurity(symbol, algorithm, marketHoursDatabase, symbolPropertiesDatabase);
+            future.Underlying = _future.Underlying;
+            return future;
+        }
+
+        /// <summary>
+        /// Determines whether or not the specified security can be removed from
+        /// this universe. This is useful to prevent securities from being taken
+        /// out of a universe before the algorithm has had enough time to make
+        /// decisions on the security
+        /// </summary>
+        /// <param name="utcTime">The current utc time</param>
+        /// <param name="security">The security to check if its ok to remove</param>
+        /// <returns>True if we can remove the security, false otherwise</returns>
+        public override bool CanRemoveMember(DateTime utcTime, Security security)
+        {
+            // if we haven't begun receiving data for this security then it's safe to remove
+            var lastData = security.Cache.GetData();
+            if (lastData == null)
+            {
+                return true;
+            }
+
+            // only remove members on day changes, this prevents us from needing to
+            // fast forward contracts continuously as price moves and out filtered
+            // contracts change thoughout the day
+            var localTime = utcTime.ConvertFromUtc(security.Exchange.TimeZone);
+            if (localTime.Date != lastData.Time.Date)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Gets the data type required for the specified combination of resolution and tick type
+        /// </summary>
+        private static Type GetDataType(Resolution resolution, TickType tickType)
```

Comment:
This method was also useful in SecurityManager.GetDataFeedType().  I refactored into LeanData and PR'd. See #680